subsys/net/lib/shell/iface.c

```diff
@@ -817,6 +818,82 @@ static int cmd_net_default_iface(const struct shell *sh, size_t argc, char *argv
 	return 0;
 }
 
+#if defined(CONFIG_ETH_PHY_DRIVER)
+static int cmd_net_link_speed(const struct shell *sh, size_t argc, char *argv[])
+{
+	int idx = get_iface_idx(sh, argv[1]);
+	const struct device *phy_dev;
+	bool half_duplex = false;
+	uint16_t user_input_spd;
+	struct net_if *iface;
+	uint16_t speed = 0;
+
+	if (argc < 3) {
+		PR_WARNING("Usage: net iface set_link <index>"
+			   "<Speed:10/100/1000/2500/5000> <Duplex[optional]:h/f>\n");
+		return -ENOEXEC;
+	}
+
+	iface = net_if_get_by_index(idx);
+	if (net_if_l2(iface) != &NET_L2_GET_NAME(ETHERNET)) {
+		PR_WARNING("Interface %d is not Ethernet type\n", idx);
+		return -EINVAL;
+	}
+
+	phy_dev = net_eth_get_phy(iface);
+	if (!phy_dev) {
+		PR_WARNING("No PHY device found for interface %d\n", idx);
+		return -ENOEXEC;
+	}
+
+	for (int k = 2; k < argc; k++) {
+		if ((k + 1 < argc) && (argv[k+1][0] == 'h')) {
+			half_duplex = true;
+		} else {
+			half_duplex = false;
+		}
+
+		user_input_spd = atoi(argv[k]);
```

Comment:
We should not use `atoi()` in the shell modules and have been trying to get rid of them. Please use `shell_strtol()` instead and check the return values so that we can catch invalid strings.